foam_controlwrapper/tests/test_controlwrapper.py

```diff
@@ -6,65 +6,306 @@
 """
 
 import unittest
-from foam_controlwrapper.foam_controlwrapper import FoamControlWrapper
+
+from simphony.cuds.mesh import Mesh, Face, Point, Cell
 from simphony.core.cuba import CUBA
+from simphony.core.data_container import DataContainer
+from simphony.io.h5_cuds import H5CUDS
+from foam_controlwrapper.foam_controlwrapper import FoamControlWrapper
 from foam_controlwrapper.cuba_extension import CUBAExt
-import shutil
-import logging
+from foam_controlwrapper.foam_files import FoamFiles
 import os
-from PyFoam.Execution.UtilityRunner import UtilityRunner
-
-logger = logging.getLogger(__name__)
+import shutil
 
 
 class FoamControlWrapperTestCase(unittest.TestCase):
     """Test case for FoamControlWrapper class"""
     def setUp(self):
-        pass
+        self.mesh = Mesh(name="mesh1")
+
+        self.points = [
+            Point(
+                (0.0, 0.0, 0.0)),
+            Point(
+                (1.0, 0.0, 0.0)),
+            Point(
+                (1.0, 1.0, 0.0)),
+            Point(
+                (0.0, 1.0, 0.0)),
+            Point(
+                (0.0, 0.0, 1.0)),
+            Point(
+                (1.0, 0.0, 1.0)),
+            Point(
+                (1.0, 1.0, 1.0)),
+            Point(
+                (0.0, 1.0, 1.0))
+        ]
+
+        puids = [self.mesh.add_point(point) for point in self.points]
+
+        self.faces = [
+            Face([puids[0], puids[3], puids[7], puids[4]],
+                 data=DataContainer({CUBA.LABEL: 0})),
+            Face([puids[1], puids[2], puids[6], puids[5]],
+                 data=DataContainer({CUBA.LABEL: 1})),
+            Face([puids[0], puids[1], puids[5], puids[4]],
+                 data=DataContainer({CUBA.LABEL: 2})),
+            Face([puids[3], puids[2], puids[6], puids[7]],
+                 data=DataContainer({CUBA.LABEL: 3})),
+            Face([puids[0], puids[1], puids[2], puids[3]],
+                 data=DataContainer({CUBA.LABEL: 4})),
+            Face([puids[4], puids[5], puids[6], puids[7]],
+                 data=DataContainer({CUBA.LABEL: 5}))
+
+        ]
+
+        [self.mesh.add_face(face) for face in self.faces]
+
+        self.cells = [
+            Cell(puids)
+        ]
+
+        self.puids = puids
+
+        [self.mesh.add_cell(cell) for cell in self.cells]
+
+    def tearDown(self):
+        FoamFiles().remove_parser_files(os.getcwd())
+
+    def test_add_mesh(self):
```

Comment:
The test_add_mesh and test_delete_mesh do not test the specifci functionalit of add_mesh or delete_mesh. They just verify that running the related functions does not raise Execptions.

For example a method defined as follows:
```python
def add_mesh(self, mesh):
      pass

```
would still pass the tests. In short the `test_add_mesh` can **never** fail can only produce an **error**.  A more appropriate name for this method would be `test_add_mesh_does_not_raise_execeptions` but this would also be uncessary as explained below. 

A proper `test_add_mesh` should be based on calling `add_mesh` and verify the calling that method had some effect. In this case (if I am correct) `add_mesh` should create a set of files in a predefined directory. A simple check then would be to **assert** that these tests have been created and have non-zero size. A more advanced check would be to verify the content of these files (but this I understand is not always trivial). To conclude, given that `test_add_mesh` verifies the effects of calling `add_mesh`, the unittest framework will signify a **failure** when assertion failed (e.g. the created files have the wrong names), but will also singify an **error**  if for some reason the test could not complete due to an unknown exception (thus a `test_add_mesh_does_not_raise_execeptions` whould be unecessary).